docker/base-runner.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+
+host_architecture() {
+    # there's numerous compatibility modes, so we want
+    # to ensure that these are valid. we also want to
+    # use dpkg if it's available since it gives hard-float
+    # information on compatible architectures
+    local host
+    local arch
+    if dpkg >/dev/null 2>&1; then
+        host=$(dpkg --print-architecture)
+        arch="${host}"
+    else
+        host=$(uname -m)
+        arch="${host}"
+
+        case "${arch}" in
+            aarch64|armv8b|armv8l)
+                arch=arm64
+                ;;
+            aarch64_be)
+                arch=arm64be
+                ;;
+            arm)
+                arch=unknown
```

Comment:
We don't know if the target is ARMv5TE, ARMv6, or ARMv7, or if it's a soft-float or hard-float target. Always default to Qemu if it's a non-Debian-based 32-bit ARM host.

Reply:
This actually needs to be `arm*`, since it can outp[ut `armv7l`, even on ARMv6 (at least according to Alpine).